Purchases/Purchasing/StoreProduct.swift

```diff
@@ -106,10 +108,28 @@ public typealias SK2Product = StoreKit.Product
 
     @objc public override var localizedTitle: String { underlyingSK2Product.displayName }
 
+    @objc public override var priceFormatter: NumberFormatter? {
+        guard let attributes = jsonDict["attributes"] as? [String: Any],
+              let offers = attributes["offers"] as? [[String: Any]],
+              let currencyCode: String = offers.first?["currencyCode"] as? String else {
+            Logger.appleError("Can't initialize priceFormatter for SK2 product! Could not find the currency code")
+            return nil
+        }
```

Comment:
I don't love this. Would love ideas for cleaning it up. 

I didn't want to go full-on with Codable since we'd only be reading one attribute, but perhaps that's nicer.

Reply:
I'd say as long as there tests the implementation is not as important. We can clean it up in the future. Maybe leave a comment to do that if we ever need to read more data?